pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -704,12 +706,23 @@ func isLRU(typ ebpf.MapType) bool {
 	return false
 }
 
-func hashMapMemoryUsage(info *ebpf.MapInfo, nrCPUS uint64) (max uint64, rss uint64) {
+func hashMapMemoryUsage(fd int, info *ebpf.MapInfo, nrCPUS uint64) (max uint64, rss uint64) {
+	prealloc := (info.Flags & unix.BPF_F_NO_PREALLOC) == 0
+
+	// memory usage is hard to approximate for non-preallocated maps, so we need to read it from fdinfo.
+	// support for precise memory footprint counting has been introduced since kernel 6.4+
+	if !prealloc && preciseMapMemUsageSupported() {
+		usage := readPreciseMapMemUsageFromFdInfo(fd)
+		if usage != 0 {
+			return usage, usage
+		}
+	}
+
```

Comment:
If the map is !prealloc, we should return 0 instead of approximating memory usage, since the approximation will likely be wrong. We do not want users drawing wrong conclusion because of incorrect data.

I realize that the old approach was emitting the approximation, but I think that was incorrect.

Reply:
Maybe as a follow up to this PR we can overload the map type tag to distinguish between prealloc and !prealloc maps as well. This would make interpreting the 0 value in the UI easier as well.

Reply:
> Maybe as a follow up to this PR we can overload the map type tag

@gjulianm expressed concerns about this potentially increasing the cardinality of ebpf-check metrics